src/dawn/engine_dawn.cc

```diff
@@ -1260,26 +1260,25 @@ Result EngineDawn::DoPatchParameterVertices(
 Result EngineDawn::DoBuffer(const BufferCommand* command) {
   Result result;
 
-  ::dawn::Buffer* dawn_buffer;
+  ::dawn::Buffer* dawn_buffer = nullptr;
+
+  const auto descriptor_set = command->GetDescriptorSet();
```

Comment:
I also did a drive-by simplification of the code around looking up the descriptorset/binding in the maps.